src/ansi-c/goto-conversion/goto_check_c.cpp

```diff
@@ -761,12 +761,13 @@ void goto_check_ct::conversion_check(const exprt &expr, const guardt &guard)
       else if(old_type.id() == ID_floatbv) // float -> signed
       {
         // Note that the fractional part is truncated!
-        ieee_floatt upper(to_floatbv_type(old_type));
+        auto rm = ieee_floatt::rounding_modet::ROUND_TO_EVEN;
+        ieee_floatt upper(to_floatbv_type(old_type), rm);
```

Comment:
Why is this not `ieee_float_valuet`?

Reply:
If the signed integer type is bigger than the floating-point type, then rounding would happen.
Arguably, you'd then not need the check.